### DISCUSSION TITLE
Matrox cards missing?

### DISCUSSION THREAD

**Discussion by shitsux420** [86Box/86Box#2044] [Q&A] [answered]
I notice that the roms folder contains a matrox folder with .vbi files for the Matrox Mystique, however this, or any other Matrox card, is not selectable as a video card in any machine configuration. Is support for Matrox cards truncated, or do you simply need the BIOS .bin files for them to work?

**Comment by tokariu** [comment, marked as answer]
I was wondering the same, but I found this in the source code:

https://github.com/86Box/86Box/blob/a2ab7467f9bdf76f07a5e8ddb4a04e58c3a491cc/src/video/vid_table.c

```
    { &s3_trio3d2x_pci_device				},
#if defined(DEV_BRANCH) && defined(USE_MGA)
    { &millennium_device					},
    { &mystique_device					},
    { &mystique_220_device				},
#endif
    { &tgui9440_pci_device				},
```

seems like matrox millennium and mystique video devices are only enabled in a development branch with USE_MGA defined.
I didn't find any dev branch or build with these cards enabled. Would be great to get more information about what's going on with these video cards. I'm also looking eager for progress on these devices. Maybe someone else can get us more information.

**Reply by TC1995** [reply]
Glitches in various ways, the Mystique/220 makes win95 guests crash non-430fx chipsets in unpredicable ways and the Millennium has transparency glitches with OS/2.

**Reply by massof971** [reply]
Hi everyone,
Is there any news about the possibility of using Matrox cards? They are still not selectable. Thank you all

**Reply by TC1995** [reply]
Too buggy for general use.

**Reply by shitsux420** [reply]
Is this something that could be improved with time or perhaps some help, like documentation or knowledge about Matrox cards?

**Reply by pankoza2-pl** [reply]
As I said, the Millenium was recently undevbranched because now it's now mostly usable

**Comment by pankoza2-pl** [comment]
the Matrox Millenium was recently undevbranched for now

**Comment by WhiteMagicRaven** [comment]
![Monitor_1_20240611-184631-274](https://github.com/86Box/86Box/assets/876874/d3a5ec68-bb34-4429-95fb-3ae4233cb3ea)
![Monitor_1_20240611-185906-729](https://github.com/86Box/86Box/assets/876874/71e44664-cddb-4d0d-ac56-4cecfbe5bc51)

sorry for bumping, i found that 2mb millennium is ok, but using 4mb or 8mb causes distortion in image on game Nascar Racing which uses MGA 3D Acceleration capabilities of this video card.

**Reply by TC1995** [reply]
bisect if it's a regression.

**Comment by tokariu** [comment]
I might as well link to another post with some kind of graphical glitches:

https://github.com/86Box/86Box/discussions/4376#discussioncomment-9348546

what's causing these tiles?
